src/polydb.jl

```diff
@@ -0,0 +1,308 @@
+   if haskey(d, "required")
+      return d["required"]
+   elseif haskey(d, "then")
+      return _read_fields(d["then"])
+   else
+      return _read_fields(d["allOf"])
+   end
+end
+
+function _print_fields(a::Array{Pair{String, String}})
+   res = ""
+   for (key, value) in a
+      if value == ""
+         res = string(res, key, "\n")
+      else
+         res = string(res, key, ": ", value, "\n")
+      end
+   end
+   return res
+end
+
+# prints information about a specific Collection
+# also used for the info(::Database) function
+function _info(io::IO, coll::Collection)
+   db = coll.mcol.database
+   coll_c = db[string("_collectionInfo.", coll.mcol.name)]
+   info = iterate(coll_c)[1]
+   print(io, typeof(coll), "\n", _get_collection(info))
+end
+
+Base.show(io::IO, coll::Collection) = _info(io, coll)
+
+Base.show(io::IO, ::MIME"text/plain", coll::Collection) = print(io, typeof(coll), ": ", coll.mcol.name)
+
+# returns an array containing the names of all collections in the Polydb, also including meta collections
+function _get_collection_names(db::Database)
+   opts = Mongoc.BSON("authorizedCollections" => true, "nameOnly" => true)
+   return Mongoc.get_collection_names(db.mdb;options=opts)
+end
+
+# returns an array cotaining the names of all collections in the Polydb, excluding meta collections
+function get_collection_names(db::Database)
+   names = _get_collection_names(db)
+   res = Array{String, 1}()
+   sizehint!(res, Int64(floor(length(names)/2)))
+   for name in names
+      if SubString(name, 1, 1) != "_"
+         push!(res, name)
+      end
+   end
+   return res
+end
+
+# for the set of names obtained by the _get_collection_names(::Database) function
+# returns two arrays containing the names of the meta data collections
+# first one for sections, second one for collections
+function _get_meta_names(names::Array{String, 1})
+   n = length(names)
+   sec_bool = BitArray{1}(undef, n)
+   coll_bool = BitArray{1}(undef, n)
+   n_secs = 0
+   n_colls = 0
+   i = 1
+   for name in names
+      if SubString(name, 1, 2) == "_s"
+         sec_bool[i] = true
+         coll_bool[i] = false
+         n_secs += 1
+      elseif SubString(name, 1, 2) == "_c"
+         sec_bool[i] = false
+         coll_bool[i] = true
+         n_colls += 1
+      else
+         sec_bool[i] = false
+         coll_bool[i] = false
+      end
+      i += 1
+   end
+   secs = Base.Array{String,1}(undef, n_secs)
+   colls = Base.Array{String,1}(undef, n_colls)
+   i_s = 1
+   i_c = 1
+   for j = 1:n
+      if sec_bool[j]
+         secs[i_s] = names[j]
+         i_s += 1
+      elseif coll_bool[j]
+         colls[i_c] = names[j]
+         i_c += 1
+      end
+   end
+   return secs, colls
+end
+
+# functions helping printing metadata for sections or collections
+function _get_contact(s::String)
+   return s
+end
+
+function _get_contact(a::Array)
+   res = ""
+   for dict in a
+      str = dict["name"]
+      for key in ["email", "www", "affiliation"]
```

Comment:
```julia
str = join(dict[key] for key in ["name", "email", "www", "affiliation"] if !isempty(get(dict, key, "")), ", ")
```
Are we sure that "name" is always non-empty?
(this is just stylistic, if You find the explicit loop version more readable, leave it as it is)

Reply:
by now, "name" always is empty. but including an additional check for future compatibility should not affect runtime noticably in this context